homeassistant/components/kitchen_sink/sensor.py

```diff
@@ -21,7 +21,8 @@
 async def async_setup_entry(
     hass: HomeAssistant,
     config_entry: ConfigEntry,
-    async_add_entities: AddEntitiesCallback,
+    # pylint: disable-next=hass-argument-type
+    async_add_entities: AddConfigEntryEntitiesCallback,
```

Comment:
Our pylint checker requires this to be `AddEntitiesCallback`, but it should be the new shiny type added in PR #133353, we could do either of:
- update the pylint rule to only allow the new callback type and update all 2900 `async_setup_entry`-functions to use the new type
- update the pylint rule to allow either callback type, if we don't want to update all integrations. This alternative is technically incorrect though, because the function passed to `async_setup_entry` *does* have the `config_subentry_id` parameter regardless of if the integration supports config subentries or not.

https://github.com/home-assistant/core/blob/9944b38cfd58c0fa84951e5350c9c81a3d0d6a1e/homeassistant/helpers/entity_platform.py#L83-L96